src/main/java/Iris.java

```diff
@@ -0,0 +1,143 @@
+import java.util.Scanner;
+
+public class Iris {
+    public static Task[] tasks = new Task[100];
+    public static int numOfTasks = 0;
+
+    public static void printDivider() {
+        System.out.println("---------------------------------------------");
+    }
+
+    public static void listTasks() {
+        if (numOfTasks == 0) {
+            System.out.println("No tasks added.");
+            return;
+        }
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < numOfTasks; i++) {
+            Task currentTask = tasks[i];
+            System.out.println((i + 1) 
+                    + "."
+                    + currentTask);
+        }
+    }
+
+    public static void addTask(String text) {
+        String[] textParts = text.split(" ", 2);
+        boolean hasNoDetails = textParts.length == 1;
+        if (hasNoDetails) {
+            System.out.println("Invalid or incomplete command");
+            return;
+        }
+
+        String command = textParts[0].toLowerCase();
+        String details = textParts[1];
+        Task newTask;
+        try {
+            switch (command) {
+            case "todo":
+                newTask = new Todo(details);
+                break;
+            case "deadline":
+                newTask = new Deadline(details);
+                break;
+            case "event":
+                newTask = new Event(details);
+                break;
+            default:
+                System.out.println("Invalid Command");
+                return;
+            }
+            tasks[numOfTasks++] = newTask;
+            printAddTaskMessage(newTask);
+        } catch (RuntimeException e) { // For missing description
+            System.out.println(e.getMessage());
+        }
+    }
+
+    private static void printAddTaskMessage(Task newTask) {
+        System.out.println("Got it. I've added this task:\n"
+                + newTask
+                + "\nNow you have "
+                + numOfTasks
+                + " tasks in the list");
+    }
+
+    public static void markTask(String text) {
+        try {
+            Task taskToMark = getTaskToEdit(text);
+            taskToMark.markAsDone();
+            System.out.println("Nice! I've marked this task as done:\n"
+                    + taskToMark);
+        }
+        catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    public static void unmarkTask(String text) {
+        try {
+            Task taskToUnmark = getTaskToEdit(text);
+            taskToUnmark.unmarkFromDone();
+            System.out.println("OK, I've marked this task as not done yet:\n"
+                    + taskToUnmark);
+        }
+        catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    private static Task getTaskToEdit(String text) {
```

Comment:
Good use of exception handling to catch missing and invalid task numbers. The code makes it clear when the user provides incomplete information.